geopandas/io/geoarrow.py

```diff
@@ -16,12 +16,62 @@ class CoordinateDimension(str, Enum):
     XYZM = "xyzm"
 
 
+def _convert_inner_coords(coords, interleaved, dims, mask=None):
+    if interleaved:
+        coords_field = pa.field(
+            "xy" if len(dims) == 2 else "xyz", pa.float64()
+        )  # , nullable=False)
+        typ = pa.list_(coords_field, len(dims))
+        parr = pa.FixedSizeListArray.from_arrays(coords.flatten(), type=typ, mask=mask)
+    else:
+        if dims == CoordinateDimension.XY:
+            parr = pa.StructArray.from_arrays(
+                [coords[:, 0].copy(), coords[:, 1].copy()], names=["x", "y"], mask=mask
```

Comment:
Do you need the `.copy()`? I would've thought this would give an extra, unneeded copy on the Python side.

Reply:
I assume it's not needed, but then the copy will happen in the pyarrow C++ conversion code anyway (because Arrow doesn't support strided data). So I think it _might_ be better to let numpy do that copy (I would not be surprised if it was faster at doing that compared to pyarrow)